views/components/ShareCharts.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from 'react'
+import Select from '@material-ui/core/Select'
+import Dialog from '@material-ui/core/Dialog'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import Chip from '@material-ui/core/Chip'
+import MenuItem from '@material-ui/core/MenuItem'
+import Typography from '@material-ui/core/Typography'
+import Button from '@material-ui/core/Button'
+import Input from '@material-ui/core/Input'
+
+import { fetchUsernames, shareChart } from '../fe-utils/fetchUtil'
+
+const MenuProps = {
+  PaperProps: {
+    style: {
+      marginTop: '50px',
+    },
+  },
+}
+
+const ShareChart = ({
+  open,
+  owner,
+  setDialogState,
+  chart,
+  getCharts,
+  setChartList,
+}) => {
+  const [users, setUsers] = useState([])
+  const [sharedList, setShared] = useState([])
+  const handleChange = (event) => setShared(event.target.value)
+  const shareWithUsers = async (chart_id, sharedWith) => {
+    try {
+      const {
+        data: { ok },
+      } = await shareChart(chart_id, sharedWith)
+
+      if (ok === 1) {
+        const { data } = await getCharts()
+        setChartList(data)
+        setDialogState({})
+      }
+    } catch (error) {}
+  }
+  const handleDelete = (userToDelete) => {
+    setShared((users) => users.filter((username) => username !== userToDelete))
+  }
+  const closeDialog = () => {
+    setDialogState({})
+  }
+  useEffect(() => {
+    fetchUsernames().then((data) => {
+      const usernames = data
+        .map((username) => ({
+          value: username,
+          label: username,
+        }))
+        .filter(({ value }) => value !== owner)
+      setUsers(usernames)
+    })
+    if (chart?.sharedWith?.length) setShared(chart.sharedWith)
```

Comment:
do we need a state for this or can we just reference `chart.sharedWith` where we need to?

Reply:
Defaulting to the chart.sharedWith list instead.
https://github.com/AMP-SCZ/dpdash/pull/131/commits/4b7a005d45471fc07236b428f61a2f58b4284d6a